vispy/visuals/_scalable_textures.py

```diff
@@ -359,18 +394,20 @@ def _get_texture_format_for_data(self, data, internalformat):
             num_channels = self._data_num_channels(data)
             texture_format = self._handle_auto_texture_format(internalformat, data)
             texture_format = self._get_gl_tex_format(texture_format, num_channels)
+        else:
+            raise NotImplementedError("wut?")
         return texture_format
 
     def _compute_clim(self, data):
         clim = self._clim
         is_auto = isinstance(clim, str) and clim == 'auto'
-        if data.ndim == 2 or data.shape[2] == 1:
+        if data.ndim == self._ndim or data.shape[2] == 1:
             if is_auto:
-                clim = np.min(data), np.max(data)
-            clim = (np.float32(clim[0]), np.float32(clim[1]))
+                clim = get_default_clim_from_data(data)
+            # clim = (np.float32(clim[0]), np.float32(clim[1]))
```

Comment:
Similar question to the CPU-based chunk of code. Shouldn't this be here? Maybe I took this from the old code and it isn't needed anymore? Or maybe it is to avoid integer operations in any of the math.